### ISSUE TITLE
Can I use the pretrained model to do inference with --no_instance flag?

### ISSUE THREAD

**Issue by aschneid42** [NVIDIA/pix2pixHD#163]
If I use the pretrained model to do inference on a new cityscapes dataset, I use this command and the results look great!
    `python test.py --name label2city_1024p --netG local --ngf 32 --resize_or_crop none`

However, if I try to do inference on this same new dataset, with the --no_instance flag,
    `python test.py --name label2city_1024p --netG local --ngf 32 --resize_or_crop none --no_instnace`,
then I get a warning saying:
> Pretrained network G has fewer layers; The following are not initialized: ['model', 'model1_1']
It will still continue to produce synthetic outputs, but they look terrible. 

Does something need to be modified to use the pretrained model for inference with --no_instance flag, or does the model need to be retrained with --no_instance in order to do inference in this way?

Thanks for your help!

**Comment by joashchn** [comment]
same question，Have you solved the problem？I trained my own RGB datasets without inst(and as readMe said, I renamed data folder to train_A, train_B, test_A)，it seems good in ./checkpoint(training intermediate results)， but when I test(with params --no_instance --label_nc 0 like training), I got same warning(Pretrained network G has fewer layers; The following are not initialized: ['model', 'model1_1']) and terrible result(actually I think it's wrong result because of wrong my config)
thx!

**Comment by joashchn** [comment]
solved it , just remove the param --ngf 32

**Comment by aschneid42** [comment]
If I use the same pretrained model, but remove the flag --ngf 32, I get a new warning:

> Pretrained network G has fewer layers; The following are not initialized: ['model', 'model1_1', 'model1_2']

The results change, but still look terrible.  This time more cartoon-like.  

But perhaps what you are saying is that I still need to re-train the model also with no instances? And also remove the --ngf 32 during training?
Thanks.

**Comment by joashchn** [comment]
> @joashchn
> If I use the same pretrained model, but remove the flag --ngf 32, I get a new warning:
> 
> > Pretrained network G has fewer layers; The following are not initialized: ['model', 'model1_1', 'model1_2']
> 
> The results change, but still look terrible. This time more cartoon-like.
> 
> But perhaps what you are saying is that I still need to re-train the model also with no instances? And also remove the --ngf 32 during training?
> Thanks.

hi，
I removed --ngf 32 because I did not use the pre-trained model, the default ngf of in base_option.py is 64,so when I test on my own training model,I copied test command line from readme include '--ngf 32' caused wrong result.
if u use pre-trained model,you should use the same ngf as pre-trained model,probably 32.

**Comment by aschneid42** [comment]
Ok, thanks for the tips.  But if you trained your own model with --no_instance, then my question is still:

Does something need to be modified to use the pretrained model for inference with --no_instance flag, or does the model need to be retrained with --no_instance in order to do inference in this way? @tcwang0509

**Comment by MuhammadAsadJaved** [comment]
> solved it , just remove the param --ngf 32

@joashchn  What kind of dataset you are generating? How are the results after training on train_A train_B?   Are you only using paired images like train_a and corresponding train_b images without label maps or instances?   

I am working on the same problem. I want to generate infrared images from visual images. I have paired images for training.  So i just want to know about results.  I have tried pix2pix before but the results was not good. So i just want to know how are the results using pix2pixHD. if it is better then i will train model otherwise will not waste time on it.  Thanks in advance

**Comment by RAYRAYRAYRita** [comment]
@aschneid42 
Hello! I was wondering the same question as you. I want to train model on cityscapes labels with instance and test my own labels without instance. So is it possible to get good results when training with instance and testing with no_instance??

**Comment by vishwas1234567** [comment]
yes but if the technique of image deep prior has to be used and you will get the accurate rading if you want to do your way.

**Comment by ljx425** [comment]
> > solved it , just remove the param --ngf 32
> 
> @joashchn What kind of dataset you are generating? How are the results after training on train_A train_B? Are you only using paired images like train_a and corresponding train_b images without label maps or instances?
> 
> I am working on the same problem. I want to generate infrared images from visual images. I have paired images for training. So i just want to know about results. I have tried pix2pix before but the results was not good. So i just want to know how are the results using pix2pixHD. if it is better then i will train model otherwise will not waste time on it. Thanks in advance

Hi! How about your work with generating infrared images from visual images? I am trying it too but can't get useful models. It's well working on the training datasets, but when I trying to generate infrared images from another datasets, the result is very bad. So I just want to know how you solved this problem in the end. Thanks in advance!

**Comment by vivekbharadhwajsa** [comment]
> Ok, thanks for the tips. But if you trained your own model with --no_instance, then my question is still:
> 
> Does something need to be modified to use the pretrained model for inference with --no_instance flag, or does the model need to be retrained with --no_instance in order to do inference in this way? @tcwang0509

Did you find a way to solve this issue?

**Comment by zhangxiaojuan66** [comment]
你好，邮件已收到，祝你万事如意，生活愉快！
